### PR TITLE
ClojureTokenScanner and fixes SyntaxColoringPreferencePage

### PR DESCRIPTION
**Pull request by arichiardi**
Hi Laurent, here is my work of today, I have isolated the coloring fix, don't ask me too many things today because my head will explode :) The git commit comment hopefully gives some better explanation and this time I added the copyright to the new files ;)

```Completely revisited the management of ClojureTokenScanner and everything related to syntax
coloring: the algorithm stays the same of course but now the scanner reacts to preferences changes
and is able to adapt its syntax coloring. This is particularly evident in
SyntaxColoringPreferencePage, which has been refactored a bit in order to keep the previewer always visible at the bottom of the key list.```

```ClojureSourceViewer, ClojureSourceViewerConfiguration and ClojureTokenScanner have become listeners to coloring preference changes, they respect configure/unconfigure life-cycle.```

```This patch also refactors the location of scanner classes (ccw.editors.clojure.scanners). ```

### CODE REVIEW

**Comment by laurentpetit** [comment]
Sounds great!
Will try to code review it ASAP!

**Comment by arichiardi** [comment]
Done with my cleaning here.

**Comment by laurentpetit** [comment]
From a usability point of view, it's a real improvement, and seems stable. Will code review now.

File: ccw.core/plugin.properties [hunk 1]
```diff
@@ -153,3 +153,5 @@ ccw.leiningen.problemmarker.classpathcontainer=Leiningen Classpath Container
 ccw.leiningen.command.launch-headless-repl.name=Launch Headless REPL for the project
 ccw.leiningen.command.launch-headless-repl.description=Launch a Headless REPL for the project through Leiningen
 
+Dummy.label=This dummy label is dedicated to all the contributors of org.eclipse.jdt.
```
Comment (laurentpetit) [inline ccw.core/plugin.properties]: What is this Dummy label for?

File: ccw.core/plugin.properties [hunk 2]
```diff
@@ -153,3 +153,5 @@ ccw.leiningen.problemmarker.classpathcontainer=Leiningen Classpath Container
 ccw.leiningen.command.launch-headless-repl.name=Launch Headless REPL for the project
 ccw.leiningen.command.launch-headless-repl.description=Launch a Headless REPL for the project through Leiningen
 
+Dummy.label=This dummy label is dedicated to all the contributors of org.eclipse.jdt.
+Presentation.label= Clojure
```
Comment (laurentpetit) [inline ccw.core/plugin.properties]: Same question maybe?

File: ccw.core/plugin.xml [hunk 3]
```diff
@@ -3399,4 +3399,14 @@
          </bundle>
       </component>
    </extension>
+   <extension
+         point="org.eclipse.ui.themes">
+         <themeElementCategory label="%Presentation.label" id="ccw.editors.clojure.presentation"/>
+      <colorDefinition
+         id="ccw.editors.clojure.colors.repl.expressionBackground"
```
Comment (laurentpetit) [inline ccw.core/plugin.xml]: can you explain how it works / what it does?

File: ccw.core/src/java/ccw/editors/clojure/ClojureColorManager.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
 
 /**
  * Java color manager.
+ * AR - Do we need this one now that everything is handled through ColorRegistry?
```
Comment (laurentpetit) [inline ccw.core/src/java/ccw/editors/clojure/ClojureColorManager.java]: maybe not, indeed

**Comment by laurentpetit** [comment]
Sounds good to me

File: ccw.core/plugin.xml (hunk 3 above)
Comment (arichiardi) [inline ccw.core/plugin.xml]: This basically adds one or more colors to your software color registry. It is the color that was in CcwPlugin and I thought of refactoring it out. The framework resolves it using its id.

File: ccw.core/src/java/ccw/editors/clojure/ClojureColorManager.java (hunk 4 above)
Comment (arichiardi) [inline ccw.core/src/java/ccw/editors/clojure/ClojureColorManager.java]: will remove then

File: ccw.core/plugin.properties (hunk 2 above)
Comment (arichiardi) [inline ccw.core/plugin.properties]: The presentation is for the extension, Dummy...well I think nothing in this patch, it was probably used in the hovers...I will remove it.

**Comment by laurentpetit** [comment]
Also waiting for your go on this one. Just to be sure

**Comment by arichiardi** [comment]
This one should be ok, rebasing now...

**Comment by arichiardi** [comment]
Ok rebased and tested, all ok!

**Comment by laurentpetit** [comment]
Ok sorry for the delay on this, the problem with AOT compilation drained all my focus these past days.

**Pull request merged**